### PR TITLE
[MBL-870] Fix GraphSchema with an Update to AppTrackingTransparency

### PR DESCRIPTION
**Pull request by msadoon**
# 📲 What and 🤔 Why

Firstly - don't be alarmed at the number of changes - that's mostly due to `bin/apollo-schema-download.sh` making automatic updates - there's only about 800 actual lines of application code that changed including tests.

There are a few things this PR covers:
1) `triggerThirdPartyEvents` `triggerCAPIEvents` mutation changes from running `bin/apollo-schema-download.sh`
2) After consent is allowed via our feature flagging tool - the `advertisingIdentifier` can be turned off via Settings - so we need to check `ATTrackingManager.trackingAuthorizationStatus` every time we use `triggerCapiEvents` or `KSRAnalytics` (Segment tracking)

# 🛠 How this was build out

### The structure of `AppTrackingTransparency`

It's all basically in `AppTrackingTransparency` which is now a `class` not a `struct` - because we want to set the `advertisingIdentifier` internally to the object. We don't want to allow it to be set outside its' own instance.

Which makes sense because we have a stack of `Environment` objects that refreshes every time the app is backgrounded/foregrounded. See `didFinishLaunchingWithOptions` `fromStorage`, `replaceEnvironment` (many places) and `pushEnvironment` (many places). So if by default we have `Environment(appTrackingTransparency: AppTrackingTransparency = AppTrackingTransparency())`, then its' easier to set that `advertisingIdentifier` on `init` than after the fact which is messy.

This is much easier in a `class` than ` struct` because a `struct` requires making functions `mutating` to change internal properties.

### Ensuring we represent the latest permissions for tracking from the system

`AppDelegateViewModel` `requestATTrackingAuthorizationStatus` has also changed to check if the app is `active` after its' launched (which is triggered when the app comes in from a background state). 

You'll also see that we call a new `updateAdvertisingIdentifier` function before checking the `advertisingIdentifier` wherever it is used. This is way to ensure the permissions are allowed before tracking right before use.

That's probably a lot to absorb and it took me about a week to figure all this out, so feel free to ask questions on any details that aren't clear. Here is the official Apple documentation where I started - https://developer.apple.com/documentation/apptrackingtransparency

### Passing around the `AppTrackingTransparency` in `AppEnvironment`

So we might use this class instead of what it was before - just the `advertisingIdentifier` string when we switch to `triggerThirdPartyEvents`. We can technically have a setting that can be sent in that mutation `applicationTrackingEnabled` that we set in-app. 

# ✅ Acceptance criteria

- [x] CI Tests are passing
- [x] First install launch is the only time system message appears if feature flag is true
- [x] Turning off tracking in Settings means the CAPI and Segment events not sent. On means they are sent. Essentially if there is an advertising identifier to send, we track users, if not we don’t.

# ⏰ TODO

- [x] Fix tests

# ⏰ Developer Testing

- [x] Launch app for first time - beta, ensure consent dialog is shown and accepted.
- [x] Browse around, go to project page, ensure segment events are being tracked, and CAPI calls are being made (try to get Charles Proxy setup for this)
- [x] Background app, go to Settings -> Privacy and Security -> Tracking -> Turn off tracking for KickBeta, go back to the app and ensure that browsing around no longer sends Segment events and if you have Charles setup - that `triggerCapiEvents` are not being sent.

### CODE REVIEW

File: KsApi/GraphAPI.swift [hunk 1]
```diff
@@ -590,9 +590,10 @@ public enum GraphAPI {
     ///   - appData
```
Comment (msadoon) [inline KsApi/GraphAPI.swift:590]: Do not review this file it's auto generated by `bin/apollo-schema-download.sh`

File: KsApi/graphql-schema.json
Comment (msadoon) [inline KsApi/graphql-schema.json:1]: Do not review this file it's auto generated by `bin/apollo-schema-download.sh`

File: Kickstarter.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (msadoon) [inline Kickstarter.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: I can't really prevent these - they are small annoying framework updates, but I generally check the release tags of the repo - if for example we go up a couple mid versions ie. 7.6.2 --> 7.8.0, to make nothing glaringly obvious has changed.

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| KsApi/models/graphql/adapters/GraphAPI.TriggerCapiEventInput+TriggerCapiEventInputTests.swift:11:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/models/graphql/adapters/GraphAPI.TriggerCapiEventInput+TriggerCapiEventInputTests.swift:12:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/models/graphql/adapters/GraphAPI.TriggerCapiEventInput+TriggerCapiEventInputTests.swift:13:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/mutations/inputs/TriggerCapiEventInputTests.swift:12:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/mutations/inputs/TriggerCapiEventInputTests.swift:13:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/mutations/inputs/TriggerCapiEventInputTests.swift:14:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/TestHelpers/TestCase.swift:41:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Library/Tracking/KSRAnalyticsTests.swift:59:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:59:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:60:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:69:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:69:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:70:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:498:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:498:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:499:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:511:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:511:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:512:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:627:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:627:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:628:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:642:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:642:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:643:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:667:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:667:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:668:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:694:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:694:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:695:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:720:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:720:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:721:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:790:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:790:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:791:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:812:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:812:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:813:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:828:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:828:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:829:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:848:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:848:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:849:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:868:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:868:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:869:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:894:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:894:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:895:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:916:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:916:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:917:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:934:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:934:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:935:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:954:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:954:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:955:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:974:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:974:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:975:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:992:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:992:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:993:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1010:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1010:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1011:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1099:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1099:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1100:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1135:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1135:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1136:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1162:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1162:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1163:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1198:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1198:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1199:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1282:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1282:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1283:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1296:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1296:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1297:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1361:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1361:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1362:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1402:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1402:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1403:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1426:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1426:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1427:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1487:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1487:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1488:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1575:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1575:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1576:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1596:1 | warning: (consecutiveBlankLines) Replace consecutive blank lines with a single blank line. |
| Library/Tracking/KSRAnalyticsTests.swift:1614:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/Tracking/KSRAnalyticsTests.swift:1632:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1632:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1633:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1726:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1726:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1727:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1756:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1756:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1757:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1766:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/Tracking/KSRAnalyticsTests.swift:1766:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| Library/Tracking/KSRAnalyticsTests.swift:1767:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by msadoon** [comment]
@scottkicks Here's a big one! Unfortunately couldn't be avoided because the schema changes. This PR fix unblocks our test failures on all the other PRs. As soon as this goes in I'll review the outstanding PRs and we can probably start that merging them tomorrow.

File: Library/MockRemoteConfigClient.swift [hunk 2]
```diff
@@ -3,6 +3,8 @@ import Foundation
 
 private class MockRemoteConfigValue: RemoteConfigValue {
   var bool = false
+
+  override var boolValue: Bool { self.bool }
```
Comment (scottkicks) [inline Library/MockRemoteConfigClient.swift:7]: Even though this change needs to be made, it is more relevant to my creator dashboard ff related [changes ](https://github.com/kickstarter/ios-oss/pull/1828/files#diff-6cdfce6a5b1a0951b6d5a3cbf5a74312978f2f18581e39a79355841b9fec8db4). Might be worth removing to maintain a clearer context

**Review by scottkicks — APPROVED** [review]
Developer testing steps look good and everything is working well 👍 

left some comments with suggestions that are not necessary for you to consider.

File: Library/MockRemoteConfigClient.swift (hunk 2 above)
Comment (msadoon) [reply Library/MockRemoteConfigClient.swift:7]: yea good call out, I noticed that too but some tests need to pass using the `MockRemoteConfigClient` and under the hood it uses the `RemoteConfigValue` which in our helper classes references `.boolValue`.

File: Library/ViewModels/PledgePaymentMethodsViewModel.swift [hunk 3]
```diff
@@ -384,10 +384,15 @@ public final class PledgePaymentMethodsViewModel: PledgePaymentMethodsViewModelT
     _ = Signal.combineLatest(project, self.viewDidLoadProperty.signal)
       .takeWhen(stripePaymentSheetDidAppear)
       .observeValues { project, _ in
-        guard project.sendMetaCapiEvents,
-          let externalId = AppEnvironment.current.advertisingIdentifier
+        guard project.sendMetaCapiEvents else { return }
+
+        AppEnvironment.current.appTrackingTransparency.updateAdvertisingIdentifier()
+
```
[the remark this answers is not in the document]
Comment (msadoon) [reply Library/ViewModels/PledgePaymentMethodsViewModel.swift:390]: Yea that's a good idea - in the cases where to do the check for CAPI events before calling `triggerCapiEvents` let's do that. Could you add that once we start on `triggerThirdPartyEvents`?

[the remark this answers is not in the document]
Comment (scottkicks) [reply Library/ViewModels/PledgePaymentMethodsViewModel.swift:390]: sounds good. i'll make a note

File: Library/MockRemoteConfigClient.swift (hunk 2 above)
Comment (scottkicks) [reply Library/MockRemoteConfigClient.swift:7]: ah right. ok cool

**Pull request merged**